backend/app/data_outputs/service.py

```diff
@@ -121,14 +122,21 @@ def remove_data_output(self, id: UUID) -> DataOutputModel:
         result = copy.deepcopy(data_output)
         self.db.delete(data_output)
         self.db.commit()
+        self.update_search_vector_associated_datasets(result)
         return result
 
+    def update_search_vector_associated_datasets(self, result: DataOutputModel):
+        dataset_service = DatasetService(self.db)
+        for dataset_link in result.dataset_links:
+            dataset_service.recalculate_search_vector_for(dataset_link.dataset_id)
```

Comment:
Maybe don't db.commit() in this function, but only in the end after everything?

Reply:
Efficiency, I believe that FastAPI even automatically commits at the end of your call.
Every call in FastAPI already runs in a transaction automatically, that is commited at the end. So probably it's not even needed.
You could use a flush instead, if you need the result in another query